packages/kas/src/nodes.ts

```diff
@@ -2817,49 +2890,52 @@ export class Eq extends Expr {
     // Assumptions: Expression is of the form a+bx, and we solve for x
     solveLinearEquationForVariable(variable) {
         var expr = this.asExpr();
-        if (!expr.is(Add) || expr.terms.length !== 2) {
+        if (!(expr instanceof Add) || expr.terms.length !== 2) {
             throw new Error(
                 "Can only handle linear equations of the form " +
                     "a + bx (= 0)",
             );
         }
 
-        var hasVar = function (term) {
+        var hasVar = function (term: Expr) {
             return term.has(Var) && _.contains(term.getVars(), variable.symbol);
         };
 
-        var a, b;
-
-        if (hasVar(expr.terms[0])) {
-            a = Mul.handleNegative(expr.terms[1]);
-            b = Mul.handleDivide(expr.terms[0], variable);
-        } else {
-            a = Mul.handleNegative(expr.terms[0]);
-            b = Mul.handleDivide(expr.terms[1], variable);
-        }
+        const [a, b] = hasVar(expr.terms[0])
+            ? [
+                  Mul.handleNegative(expr.terms[1]),
+                  Mul.handleDivide(expr.terms[0], variable),
+              ]
+            : [
+                  Mul.handleNegative(expr.terms[0]),
+                  Mul.handleDivide(expr.terms[1], variable),
+              ];
 
         return Mul.handleDivide(a, b).simplify();
     }
 }
 
 /* abstract symbol node */
-class Symbol extends Expr {
+class Sym extends Expr {
     needsExplicitMul() {
         return false;
     }
 
-    findGCD(factor) {
-        if (factor instanceof Symbol || factor instanceof Num) {
-            return this.equals(factor) ? this : Num.One;
+    findGCD(factor: Readonly<Expr>): Expr {
```

Comment:
There were a few places that I made params `Readonly<>`.  This was an attempt to fix some issues with TS thinking that subclasses weren't assignable to a superclass.  This change probably isn't necessary so I'm going to try removing.  In general though, I think making things `Readonly<>` by default I preferable.  Maybe that could happening in a more comprehensive way in a future PR.